src/interpreter/runtime/castIntoStackLocalOfType.js

```diff
@@ -9,13 +9,16 @@ const f64 = require("./values/f64");
 export function castIntoStackLocalOfType(
   type: string,
   v: any,
-  nan: boolean = false
+  nan: boolean = false,
+  inf: boolean = false
 ): StackLocal {
   const castFn = {
     i32: i32.createValueFromAST,
     i64: i64.createValueFromAST,
-    f32: nan ? f32.createNanFromAST : f32.createValueFromAST,
-    f64: f64.createValueFromAST
+    f32: inf
+      ? f32.createInfFromAST
+      : nan ? f32.createNanFromAST : f32.createValueFromAST,
```

Comment:
I'm not a super fan of terner in terner :(

Reply:
Good point, I thought maybe we could pass the whole AST node into `createValueFromAST`, so I dont need to pass `inf` and `nan` flags around and everything is handled inside one method. What do you think?

Reply:
Yeah, like `AST Node To StackLocal` :+1:

Reply:
I replaced the ternaries with if statements for now, changing the signature of `createValueFromAST` will be a bigger refactor which I'd like to do in a separate pr.